extension/index.js

```diff
@@ -96,6 +98,28 @@ async function openDropdown({currentTarget: indicator}) {
 		const container = select('.NPG-dropdown', dropdown);
 		empty(container);
 		container.append(...list);
+
+		const forms = select.all(('.NPG-dropdown form'));
+		for (const form of forms) {
+			delegate(form, 'button', 'click', async event => {
+				event.preventDefault();
+				await postForm(event.delegateTarget.form);
+				const svgIconClassList = select('svg', event.delegateTarget).classList;
+				const issueNotificationParent = event.delegateTarget.closest('.issue-notification');
+
+				if (svgIconClassList.contains('octicon-primitive-dot')) {
+					issueNotificationParent.classList.replace('read', 'unread');
+				} else if (svgIconClassList.contains('octicon-check')) {
+					issueNotificationParent.classList.replace('unread', 'read');
+				} else if (svgIconClassList.contains('octicon-mute')) {
+					issueNotificationParent.classList.remove('unread');
+					issueNotificationParent.classList.add('muted', 'read');
+				} else if (svgIconClassList.contains('octicon-unmute')) {
+					issueNotificationParent.classList.remove('muted');
+				}
```

Comment:
Perhaps this, slightly shorter and relies on the form itself rather than the icon.

```js
const {form} = event.delegateTarget;
await postForm(form);

const notification = form.closest('.issue-notification');
if (form.matches('.js-mark-notification-as-read, .js-mute-notification')) {
	notification.classList.replace('unread', 'read');
} else if (form.matches('.js-mark-notification-as-unread')) {
	notification.classList.replace('read', 'unread');
}

if (form.matches('.js-mute-notification')) {
	notification.classList.add('muted');
} else if (form.matches('.js-unmute-notification')) {
	notification.classList.remove('muted');
}
```

Reply:
Did so, it definitely is a better way. 

Also, changed the following
```js
if (form.matches('.js-mute-notification')) {
	notification.classList.add('muted');
```
to be
```js
if (form.matches('.js-mute-notification')) {
	notification.classList.add('muted', 'read');
```
to make it consistent with how it works on `/notifications` page when we mute a notification